### PR TITLE
[MNG-5910] Fixed the description of the POM reference & added an exce…

### PR DESCRIPTION
**Pull request by rajivpjs**
Please review and let me know any changes.

Thanks.

### CODE REVIEW

File: maven-model-builder/src/test/java/org/apache/maven/model/building/ComplexActivationTest.java [hunk 1]
```diff
@@ -56,5 +56,26 @@ public void testAndConditionInActivation()
...
+
+        ModelBuilder builder = new DefaultModelBuilderFactory().newInstance();
+        assertNotNull( builder );
+
+        DefaultModelBuildingRequest request = new DefaultModelBuildingRequest();
+        request.setProcessPlugins( true );
+        request.setPomFile( getPom( "complexEmptyAndMissing" ) );
+        request.setSystemProperties( sysProperties );
+
+        try {
```
Comment (khmarbaise) [inline maven-model-builder/src/test/java/org/apache/maven/model/building/ComplexActivationTest.java]: As far as I know we use JUnit 4.X where you can @Rule for exceptions ?

Comment (rajivpjs) [inline maven-model-builder/src/test/java/org/apache/maven/model/building/ComplexActivationTest.java]: I will work on this and have this done shortly. What do you mean by WDYT?

Comment (khmarbaise) [inline maven-model-builder/src/test/java/org/apache/maven/model/building/ComplexActivationTest.java]: Hi, WDYT means What Do You Think ? I wanted that you shared your opinion.....

Comment (rajivpjs) [inline maven-model-builder/src/test/java/org/apache/maven/model/building/ComplexActivationTest.java]: oh ok!!!

I think it is a great idea. Usually, when I want to test for Exceptions I use annotation approach like having @Test(expected = Exception.class). However, looking at the existing examples of Apache I couldn't find any. So, I actually learnt the Rule approach is kind of equivalent. I am just applying the change.

Thanks.

**Comment by khmarbaise** [comment]
The @Test (expected =...) approach has one drawback, cause you can't check the message of the exception....Would be great...

**Comment by rajivpjs** [comment]
I agree.

I have applied the following to the test class:

@Rule
	private ExpectedException expectedException = ExpectedException.none();

Within the test I have added:

expectedException.expect(ModelBuildingException.class);
    	expectedException.expectMessage("Failed due to exists and missing properties are enabled for profile exists-missing-condition");
    	builder.build( request );

However, the expectedException is not been triggered. Whatever, I do to expectedException nothing seems to trigger the Exception. Do you know if I am missing anything?

**Comment by rajivpjs** [comment]
@khmarbaise - I have it working now. Please see changes. This works by enabling @Test annotations together with @Rule.

**Comment by rajivpjs** [comment]
@khmarbaise - is the issue solved? Can this be merged and this pull request closed?

File: maven-model-builder/src/test/resources/poms/factory/complexEmptyAndMissing.xml [hunk 2]
```diff
@@ -31,21 +31,5 @@
...
-                <file>
-                	<exists>simple.xml</exists>
-                    <missing>simple.xml</missing>
-                </file>
-            </activation>
-            <properties>
-                <profile.miss>activated-2</profile.miss>
-            </properties>
-        </profile>
     </profiles>
```
Comment (khmarbaise) [inline maven-model-builder/src/test/resources/poms/factory/complexEmptyAndMissing.xml]: Can you tell me why you have removed those part of the file and changed the id's ? I don't see any good reason to remove those parts and renaming the id's ?

Comment (rajivpjs) [inline maven-model-builder/src/test/resources/poms/factory/complexEmptyAndMissing.xml]: Well spotted. Sorry, this was accidental. I will revert this change shortly.

**Comment by rajivpjs** [comment]
Hi Karl

I have now made further changes. Hopefully, it should be fine.

Please review.

Thanks.

**Comment by rajivpjs** [comment]
I know that you are busy. Could you please let me know whether there are any more changes to be done? Thanks.

**Comment by rajivpjs** [comment]
@hboutemy - changes were made as per discussion. Please review and sign off.

Thanks.

**Comment by rajivpjs** [comment]
Will reopen issue if someone does take a look. Closing issue.

**Comment by gnodet** [comment]
Superseded by https://github.com/apache/maven/pull/1773

**Pull request closed without merging**